src/components/BalanceService.tsx

```diff
@@ -0,0 +1,23 @@
+import MemoryService from "../core/Memory";
+
+class BalanceService {
+    saveValues: (balances: { date: Date, value: number }[]) => void;
+    balances: { date: Date, value: number }[] ;
+
+    addBalance(valueFromBalanceForm: number) {
+        let valuesDatesFromMemory = MemoryService.instance.getValues();
+        let valueWithDate = {date: new Date(), value: valueFromBalanceForm};
+        let valueToSave = [valueWithDate, ...valuesDatesFromMemory];
+        this.balances = [valueWithDate, ...valuesDatesFromMemory];
+        MemoryService.instance.saveValues(valueToSave);
+    }
+
+    constructor() {
```

Comment:
MemoryService should be passed through constructor to allow us to modify implementations in the future

Reply:
Do you mean: "constructor(){ this.memoryService = MemoryService.instance;}?

Reply:
rather constructor(memoryService: MemoryService).
In the future we will be changing MemoryService implementation so it would be nice to do it without violating sOlid principle.